core/ir/opnd_shared.c

```diff
@@ -2805,7 +2805,18 @@ opnd_size_t
 reg_get_size_lmul(reg_id_t reg, lmul_t lmul)
 {
     if (reg >= DR_REG_VR0 && reg <= DR_REG_VR31) {
-        int power = ((int)lmul << 29) << 29;
+        /* lmul is a 3-bit signed number encoded as the shift amount:
+         * 1/8: -3
+         * 1/4: -2
+         * 1/2: -1
+         *   0:  0
+         *   1:  1
+         *   2:  2
+         *   3:  3
+         * so (vlen >> (3 - lmul)) converts the hardware vector length in bits to the
+         * effective vector length in bytes.
+         */
+        int power = ((int)lmul << 29) >> 29;
```

Comment:
OK so this is shifting left by 29 in order to set the top bit; it then shifts back down and seems to assume that this right-shift of a negative number will sign-extend.  But that is implementation-defined so it's not guaranteed it will do what you want.  Probably it will in most implementations (seems to in GA CI as the unit test passes).

Could we do something safer: have the enum for RV64_LMUL_1_8 be -3 instead of 5?  Then we don't have to shift it at all.

Reply:
Done.